app/src/main/java/org/simple/clinic/contactpatient/ContactPatientModel.kt

```diff
@@ -68,7 +67,7 @@ data class ContactPatientModel(
     get() = appointment != null
 
   val hasRegisteredFacility: Boolean
-    get() = appointment?.get()?.patientRegisteredFacilityID != null
+    get() = appointment?.isPresent() == true && appointment.get().patientRegisteredFacilityID != null
```

Comment:
Is there a need for the `== true` here since `isPresent()` will return `Boolean`?

Reply:
So isPresent will only return a boolean if the appointment value is not null, since this is a ParcelableOptional object so if it's null we would wanna send false. 
`appointment?.isPresent() == false` if it returns null.

Reply:
No, what I meant was `appointment?.isPresent()` _should be_ functionally the same as `appointment?,isPresent() == true`. 

However, the reason it seems to be required here is because `appointment` is nullable, so the type of the expression `appointment?.isPresent()` is inferred to be `Boolean?` and not `Boolean`.